src/redis-cli.c

```diff
@@ -418,92 +425,273 @@ static sds cliVersion(void) {
+    if (tokenPart[0] != '\0' && strcmp(type, "pure-token") != 0) {
+        tokenPart = sdscat(tokenPart, " ");
+    }
+    if (optional) {
+        params = sdscat(params, "[");
+    }
+    params = sdscat(params, tokenPart);
+    params = sdscat(params, repeatPart);
+    if (multiple) {
+        params = sdscat(params, " [");
+        if (multipleToken) {
+            params = sdscat(params, tokenPart);
+        }
+        params = sdscat(params, repeatPart);
+        params = sdscat(params, " ...]");
+    }
+    if (optional) {
+        params = sdscat(params, "]");
+    }
+    sdsfree(tokenPart);
+    sdsfree(repeatPart);
+    return params;
+}
+
+static void cliCreateHelpEntry(helpEntry *help, char *cmdname, char *subcommandname) {
+    help->argc = subcommandname ? 2 : 1;
+    help->argv = zmalloc(sizeof(sds) * help->argc);
+    help->argv[0] = sdsnew(cmdname);
+    sdstoupper(help->argv[0]);
+    if (subcommandname) {
+        /* Subcommand name is two words separated by a pipe character. */
+        help->argv[1] = sdsnew(strchr(subcommandname, '|') + 1);
+        sdstoupper(help->argv[1]);
+    }
+    sds fullname = sdsnew(help->argv[0]);
+    if (subcommandname) {
+        fullname = sdscat(fullname, " ");
+        fullname = sdscat(fullname, help->argv[1]);
+    }
+    help->full = fullname;
+    help->type = CLI_HELP_COMMAND;
+
+    struct commandHelp *ch = zmalloc(sizeof(*ch));
+    help->org = ch;
+    ch->name = help->full;
+    ch->params = sdsempty();
+}
+
+/* Initialize a command help entry for the command/subcommand described in specs.
+ * Returns the index of the next available position in the help entries table. */
+static helpEntry *cliInitCommandHelpEntry(char *cmdname, char *subcommandname, helpEntry *next, redisReply *specs) {
+    helpEntry *help = next++;
+    cliCreateHelpEntry(help, cmdname, subcommandname);
+
+    assert(specs->type == (config.resp3 ? REDIS_REPLY_MAP : REDIS_REPLY_ARRAY));
+    for (size_t j = 0; j < specs->elements; j += 2) {
+        assert(specs->element[j]->type == REDIS_REPLY_STRING);
+        char *key = specs->element[j]->str;
+        if (!strcmp(key, "summary")) {
+            redisReply *reply = specs->element[j + 1];
+            assert(reply->type == REDIS_REPLY_STRING);
+            help->org->summary = sdsnew(reply->str);
+        }
+        else if (!strcmp(key, "since")) {
+            redisReply *reply = specs->element[j + 1];
+            assert(reply->type == REDIS_REPLY_STRING);
+            help->org->since = sdsnew(reply->str);
+        }
+        else if (!strcmp(key, "group")) {
+            redisReply *reply = specs->element[j + 1];
+            assert(reply->type == REDIS_REPLY_STRING);
+            help->org->group = sdsnew(reply->str);
+        }
+        else if (!strcmp(key, "arguments")) {
+            redisReply *args = specs->element[j + 1];
+            assert(args->type == (config.resp3 ? REDIS_REPLY_SET : REDIS_REPLY_ARRAY));
```

Comment:
FYI: Under RESP3, the "arguments" field appears to be returned as a SET. This seems odd, as command arguments are ordered.

Reply:
@guybe7 can you please make a PR to fix this?

Reply:
https://github.com/redis/redis/pull/10188

Reply:
@jhelbaum this one is merged, so you can merge from unstable and fix the parsing code if needed.